tf/tfprocess.py

```diff
@@ -113,19 +113,30 @@ def __init__(self, cfg):
         self.renorm_max_d = self.cfg['training'].get('renorm_max_d', 0)
         self.renorm_momentum = self.cfg['training'].get('renorm_momentum', 0.99)
 
-        gpus = tf.config.experimental.list_physical_devices('GPU')
-        tf.config.experimental.set_visible_devices(gpus[self.cfg['gpu']], 'GPU')
-        tf.config.experimental.set_memory_growth(gpus[self.cfg['gpu']], True)
+        if self.cfg['gpu'] == 'all':
+            self.strategy = tf.distribute.MirroredStrategy()
```

Comment:
Replace this line with:
            self.strategy = tf.distribute.MirroredStrategy(cross_device_ops=tf.distribute.ReductionToOneDevice())
to get a functional version (at least on my windows machine ...)